action/protocol/staking/protocol.go

```diff
@@ -624,6 +624,7 @@ func (p *Protocol) settleAction(
 	if err != nil {
 		return nil, err
 	}
+	acc.ConvertFreshAccountToZeroNonceType(actionCtx.Nonce)
```

Comment:
should we change account type before hard fork?

Reply:
this is safe b/c this would only happen after the hard-fork when the nonce of clean address is changed by `AdjustNonceForFreshAddress()` to 0. Before the hard-fork, the first nonce of a fresh address remains == 1, so this does no harm

we can add the hard-fork flag to make the intention clear, and after the hard-fork remove the flag check